File: MiniDOMWebCtrls/DMD/Luoghi/LuoghiUtils.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.WebSite

Imports minidom.Forms
Imports minidom.Anagrafica
Imports minidom.Databases
Imports minidom.Forms.Utils

Namespace Forms

    Partial Public Class Utils

        Public NotInheritable Class CLuoghiUtilsClass

            Friend Sub New()
            End Sub

            Public Function CreateElencoPuntiOperativi(ByVal selItem As String) As String
                Dim writer As New System.Text.StringBuilder
                Dim i As Integer
                Dim Uffici As CCollection(Of CUfficio)
                Dim ufficio As CUfficio
                Uffici = Anagrafica.Uffici.GetPuntiOperativi
                For i = 0 To Uffici.Count - 1
                    ufficio = Uffici.Item(i)
                    writer.Append("<option value=""")
                    writer.Append(GetID(ufficio))
                    writer.Append(""" ")
                    If (GetID(ufficio) = selItem) Then writer.Append("selected")
                    writer.Append(">")
                    writer.Append(Strings.HtmlEncode(ufficio.Nome))
                    writer.Append("</option>")
                Next

                Return writer.ToString
            End Function

            Public Function CreateElencoComuni(ByVal provincia As String, ByVal comune As String) As String
                Dim ret As New System.Text.StringBuilder
                Dim t1 = False, t As Boolean = False
                comune = Strings.Trim(comune)
                provincia = Strings.Trim(provincia)
                For Each p As CComune In Luoghi.Comuni.LoadAll
                    If (p.Stato <> ObjectStatus.OBJECT_VALID) Then Continue For
                    If (Strings.Compare(p.Provincia, provincia) <> 0 AndAlso Strings.Compare(p.Sigla, provincia) <> 0) Then Continue For
                    t = Strings.Compare(p.Nome, comune, CompareMethod.Text) = 0
                    t1 = t1 OrElse t
                    ret.Append("<option value=""")
                    ret.Append(Strings.HtmlEncode(p.Nome))
                    ret.Append(""" ")
                    If (t) Then ret.Append("selected")
                    ret.Append(">")
                    ret.Append(Strings.HtmlEncode(p.Nome))
                    ret.Append("</option>")
                Next
                If (comune <> "" AndAlso Not t1) Then
                    ret.Append("<option value=""")
                    ret.Append(Strings.HtmlEncode(comune))
                    ret.Append(""" selected style=""color:red"">")
                    ret.Append(Strings.HtmlEncode(comune))
                    ret.Append("</option>")
                End If

                Return ret.ToString
            End Function




            Public Function CreateElencoNomiProvince(ByVal selValue As String) As String
                Dim ret As New System.Text.StringBuilder
                Dim t1 = False, t As Boolean = False
                selValue = Strings.Trim(selValue)
                For Each p As CProvincia In Luoghi.Province.LoadAll
                    If (p.Stato <> ObjectStatus.OBJECT_VALID) Then Continue For
                    t = Strings.Compare(p.Nome, selValue, CompareMethod.Text) = 0
                    t1 = t1 OrElse t
                    ret.Append("<option value=""")
                    ret.Append(Strings.HtmlEncode(p.Nome))
                    ret.Append(""" ")
                    If (t) Then ret.Append("selected")
                    ret.Append(">")
                    ret.Append(Strings.HtmlEncode(p.Nome))
                    ret.Append("</option>")
                Next
                If (selValue <> "" AndAlso Not t1) Then
                    ret.Append("<option value=""")
                    ret.Append(Strings.HtmlEncode(selValue))
                    ret.Append(""" selected style=""color:red"">")
                    ret.Append(Strings.HtmlEncode(selValue))
                    ret.Append("</option>")
                End If

                Return ret.ToString
            End Function

            Public Function CreateElencoSigleProvince(ByVal selValue As String) As String
                Dim ret As New System.Text.StringBuilder
                Dim t1 = False, t As Boolean = False
                selValue = Strings.Trim(selValue)
                For Each p As CProvincia In Luoghi.Province.LoadAll
                    If (p.Stato <> ObjectStatus.OBJECT_VALID) Then Continue For
                    t = Strings.Compare(p.Sigla, selValue, CompareMethod.Text) = 0
                    t1 = t1 OrElse t
                    ret.Append("<option value=""" & Strings.HtmlEncode(p.Sigla) & """ " & CStr(IIf(t, "selected", "")) & ">" & Strings.HtmlEncode(p.Sigla) & "</option>")
                Next
                If (selValue <> "" AndAlso Not t1) Then
                    ret.Append("<option value=""" & Strings.HtmlEncode(selValue) & """ " & CStr(IIf(t, "selected", "")) & " style=""color:red"">" & Strings.HtmlEncode(selValue) & "</option>")
                End If
                Return ret.ToString
            End Function

            Public Function CreateElencoNomiRegioni(ByVal selValue As String) As String
                Dim ret As New System.Text.StringBuilder
                Dim t1 = False, t As Boolean = False
                selValue = Strings.Trim(selValue)
                For Each p As CRegione In Luoghi.Regioni.LoadAll
                    If (p.Stato <> ObjectStatus.OBJECT_VALID) Then Continue For
                    t = Strings.Compare(p.Nome, selValue, CompareMethod.Text) = 0
                    t1 = t1 OrElse t
                    ret.Append("<option value=""" & Strings.HtmlEncode(p.Nome) & """ " & CStr(IIf(t, "selected", "")) & ">" & Strings.HtmlEncode(p.Nome) & "</option>")
                Next
                If (selValue <> "" AndAlso Not t1) Then
                    ret.Append("<option value=""" & Strings.HtmlEncode(selValue) & """ " & CStr(IIf(t, "selected", "")) & " style=""color:red"">" & Strings.HtmlEncode(selValue) & "</option>")
                End If
                Return ret.ToString
            End Function

        End Class

        Private Shared m_LuoghiUtils As CLuoghiUtilsClass = Nothing

        Public Shared ReadOnly Property LuoghiUtils As CLuoghiUtilsClass
            Get
                If (m_LuoghiUtils Is Nothing) Then m_LuoghiUtils = New CLuoghiUtilsClass
                Return m_LuoghiUtils
            End Get
        End Property


    End Class


End Namespace

File: MiniDOMWebCtrls/DMD/Luoghi/LuoghiUtils.cs
using DMD;
using DMD.XML;

namespace minidom.Forms
{
    public partial class Utils
    {
        public sealed class CLuoghiUtilsClass
        {
            internal CLuoghiUtilsClass()
            {
            }

            public string CreateElencoPuntiOperativi(string selItem)
            {
                var writer = new System.Text.StringBuilder();
                int i;
                CCollection<Anagrafica.CUfficio> Uffici;
                Anagrafica.CUfficio ufficio;
                Uffici = Anagrafica.Uffici.GetPuntiOperativi();
                var loopTo = Uffici.Count - 1;
                for (i = 0; i <= loopTo; i++)
                {
                    ufficio = Uffici[i];
                    writer.Append("<option value=\"");
                    writer.Append(Databases.GetID(ufficio));
                    writer.Append("\" ");
                    if (Databases.GetID(ufficio) == DMD.Doubles.CDbl(selItem))
                        writer.Append("selected");
                    writer.Append(">");
                    writer.Append(DMD.WebUtils.HtmlEncode(ufficio.Nome));
                    writer.Append("</option>");
                }

                return writer.ToString();
            }

            public string CreateElencoComuni(string provincia, string comune)
            {
                var ret = new System.Text.StringBuilder();
                bool t1 = false;
                bool t = false;
                comune = DMD.Strings.Trim(comune);
                provincia = DMD.Strings.Trim(provincia);
                foreach (Anagrafica.CComune p in Anagrafica.Luoghi.Comuni.LoadAll())
                {
                    if (p.Stato != Databases.ObjectStatus.OBJECT_VALID)
                        continue;
                    if (
                        DMD.Strings.Compare(p.Provincia, provincia, true) != 0 
                        && 
                        DMD.Strings.Compare(p.Sigla, provincia, true) != 0
                        )
                        continue;

                    t = DMD.Strings.Compare(p.Nome, comune, true) == 0;
                    t1 = t1 || t;
                    ret.Append("<option value=\"");
                    ret.Append(DMD.WebUtils.HtmlEncode(p.Nome));
                    ret.Append("\" ");
                    if (t)
                        ret.Append("selected");
                    ret.Append(">");
                    ret.Append(DMD.WebUtils.HtmlEncode(p.Nome));
                    ret.Append("</option>");
                }

                if (!string.IsNullOrEmpty(comune) && !t1)
                {
                    ret.Append("<option value=\"");
                    ret.Append(DMD.WebUtils.HtmlEncode(comune));
                    ret.Append("\" selected style=\"color:red\">");
                    ret.Append(DMD.WebUtils.HtmlEncode(comune));
                    ret.Append("</option>");
                }

                return ret.ToString();
            }

            public string CreateElencoNomiProvince(string selValue)
            {
                var ret = new System.Text.StringBuilder();
                bool t1 = false;
                bool t = false;
                selValue = DMD.Strings.Trim(selValue);
                foreach (Anagrafica.CProvincia p in Anagrafica.Luoghi.Province.LoadAll())
                {
                    if (p.Stato != Databases.ObjectStatus.OBJECT_VALID)
                        continue;
                    t = DMD.Strings.Compare(p.Nome, selValue, true) == 0;
                    t1 = t1 || t;
                    ret.Append("<option value=\"");
                    ret.Append(DMD.WebUtils.HtmlEncode(p.Nome));
                    ret.Append("\" ");
                    if (t)
                        ret.Append("selected");
                    ret.Append(">");
                    ret.Append(DMD.WebUtils.HtmlEncode(p.Nome));
                    ret.Append("</option>");
                }

                if (!string.IsNullOrEmpty(selValue) && !t1)
                {
                    ret.Append("<option value=\"");
                    ret.Append(DMD.WebUtils.HtmlEncode(selValue));
                    ret.Append("\" selected style=\"color:red\">");
                    ret.Append(DMD.WebUtils.HtmlEncode(selValue));
                    ret.Append("</option>");
                }

                return ret.ToString();
            }

            public string CreateElencoSigleProvince(string selValue)
            {
                var ret = new System.Text.StringBuilder();
                bool t1 = false;
                bool t = false;
                selValue = DMD.Strings.Trim(selValue);
                foreach (Anagrafica.CProvincia p in Anagrafica.Luoghi.Province.LoadAll())
                {
                    if (p.Stato != Databases.ObjectStatus.OBJECT_VALID)
                        continue;
                    t = DMD.Strings.Compare(p.Sigla, selValue, true) == 0;
                    t1 = t1 || t;
                    ret.Append("<option value=\"" + DMD.WebUtils.HtmlEncode(p.Sigla) + "\" " + Sistema.IIF(t, "selected", "") + ">" + DMD.WebUtils.HtmlEncode(p.Sigla) + "</option>");
                }

                if (!string.IsNullOrEmpty(selValue) && !t1)
                {
                    ret.Append("<option value=\"" + DMD.WebUtils.HtmlEncode(selValue) + "\" " + Sistema.IIF(t, "selected", "") + " style=\"color:red\">" + DMD.WebUtils.HtmlEncode(selValue) + "</option>");
                }

                return ret.ToString();
            }

            public string CreateElencoNomiRegioni(string selValue)
            {
                var ret = new System.Text.StringBuilder();
                bool t1 = false;
                bool t = false;
                selValue = DMD.Strings.Trim(selValue);
                foreach (Anagrafica.CRegione p in Anagrafica.Luoghi.Regioni.LoadAll())
                {
                    if (p.Stato != Databases.ObjectStatus.OBJECT_VALID)
                        continue;
                    t = DMD.Strings.Compare(p.Nome, selValue, true) == 0;
                    t1 = t1 || t;
                    ret.Append("<option value=\"" + DMD.WebUtils.HtmlEncode(p.Nome) + "\" " + Sistema.IIF(t, "selected", "") + ">" + DMD.WebUtils.HtmlEncode(p.Nome) + "</option>");
                }

                if (!string.IsNullOrEmpty(selValue) && !t1)
                {
                    ret.Append("<option value=\"" + DMD.WebUtils.HtmlEncode(selValue) + "\" " + Sistema.IIF(t, "selected", "") + " style=\"color:red\">" + DMD.WebUtils.HtmlEncode(selValue) + "</option>");
                }

                return ret.ToString();
            }
        }

        private static CLuoghiUtilsClass m_LuoghiUtils = null;

        public static CLuoghiUtilsClass LuoghiUtils
        {
            get
            {
                if (m_LuoghiUtils is null)
                    m_LuoghiUtils = new CLuoghiUtilsClass();
                return m_LuoghiUtils;
            }
        }
    }
}